runelite-client/src/main/java/net/runelite/client/plugins/skillcalculator/SkillCalculator.java

```diff
@@ -353,8 +355,35 @@ private void calculate()
 			{
 				actionCount = (int) Math.ceil(neededXP / xp);
 			}
+			int actionCost = 0;
+			int finalCost;
+			String materials = "";
+			// check whether or not this skill action has a material cost field in the resource json file.
+			if (action.getMaterials() != null)
+			{
+				for (Material material : action.getMaterials())
+				{
+					if (material != null)
+					{
+						actionCost += (itemManager.getItemPrice(material.getId()) * material.getAmount());
```

Comment:
`getItemPrice` needs to be called on the client thread

Reply:
Injected client thread in the plugin and passed it via constructors.